[Flang][OpenMP] Parser rejects valid OpenMP 5.0 combined construct `!$omp parallel loop`

Flang fails to parse the OpenMP 5.0 combined construct `!$omp parallel loop`, emitting a parse error. The same source compiles with CCE Fortran, gfortran, ifort, and ifx.

Repro: https://flang.godbolt.org/z/oG5od3Yf4

```fortran
subroutine s
  !$omp parallel loop
  do i = 1, 1
  end do
end subroutine

program test
  call s
end program test
```

Problem:
```bash
$ flang -fopenmp example.f90
error: Could not parse /app/example.f90
example.f90:2:22: error: expected OpenMP construct
    !$omp parallel loop
                       ^
```

AFAIU, the directive should parse successfully: `parallel loop` is a valid combined construct since OpenMP 5.0 (Section 2.13.2, `parallel loop` Construct, <https://www.openmp.org/wp-content/uploads/OpenMP-API-Specification-5.0.pdf>).

Workaround:

Splitting the construct does compile:
```fortran
!$omp parallel
!$omp loop
do i = 1, 1
end do
!$omp end loop
!$omp end parallel
```

Alternatively, `!$omp parallel do` works but is not semantically identical.

Note:
- NVFortran also fails on this pattern (internal compiler error), suggesting incomplete frontend support across LLVM-based Fortran compilers.
- I am willing to work on this if no one else is already working on this (from searching through the opened issues it seems not) so please feel free to assign this to me.

--- a/flang/lib/Parser/openmp-parsers.cpp
+++ b/flang/lib/Parser/openmp-parsers.cpp
@@ -2530,6 +2530,7 @@ static constexpr DirectiveSet GetLoopDirectives() {
       unsigned(Directive::OMPD_master_taskloop_simd),
       unsigned(Directive::OMPD_parallel_do),
       unsigned(Directive::OMPD_parallel_do_simd),
+      unsigned(Directive::OMPD_parallel_loop),
       unsigned(Directive::OMPD_parallel_masked_taskloop),
       unsigned(Directive::OMPD_parallel_masked_taskloop_simd),
       unsigned(Directive::OMPD_parallel_master_taskloop),

--- a/flang/test/Parser/OpenMP/parallel-loop-unparse.f90
+++ b/flang/test/Parser/OpenMP/parallel-loop-unparse.f90
@@ -1,0 +1,69 @@
+! RUN: %flang_fc1 -fdebug-unparse -fopenmp -fopenmp-version=50 %s | \
+! RUN:   FileCheck --ignore-case %s
+
+! RUN: %flang_fc1 -fdebug-dump-parse-tree -fopenmp -fopenmp-version=50 %s | \
+! RUN:   FileCheck --check-prefix="PARSE-TREE" %s
+
+! Check for parsing of parallel loop combined construct (OpenMP 5.0, 2.13.2)
+
+subroutine test_parallel_loop
+  integer :: i, j = 1
+  !PARSE-TREE: OmpBeginLoopDirective
+  !PARSE-TREE-NEXT: OmpDirectiveName -> llvm::omp::Directive = parallel loop
+  !CHECK: !$omp parallel loop
+  !$omp parallel loop
+  do i=1,10
+   j = j + 1
+  end do
+end subroutine
+
+subroutine test_parallel_loop_with_end
+  integer :: i, j = 1
+  !PARSE-TREE: OmpBeginLoopDirective
+  !PARSE-TREE-NEXT: OmpDirectiveName -> llvm::omp::Directive = parallel loop
+  !CHECK: !$omp parallel loop
+  !$omp parallel loop
+  do i=1,10
+   j = j + 1
+  end do
+  !PARSE-TREE: OmpEndLoopDirective
+  !PARSE-TREE-NEXT: OmpDirectiveName -> llvm::omp::Directive = parallel loop
+  !CHECK: !$omp end parallel loop
+  !$omp end parallel loop
+end subroutine
+
+subroutine test_parallel_loop_with_clauses
+  integer :: i, j = 1
+  !PARSE-TREE: OmpBeginLoopDirective
+  !PARSE-TREE-NEXT: OmpDirectiveName -> llvm::omp::Directive = parallel loop
+  !CHECK: !$omp parallel loop num_threads(4_4) collapse(1_4) private(j) default(shared)
+  !$omp parallel loop num_threads(4) collapse(1) private(j) default(shared)
+  do i=1,10
+   j = j + 1
+  end do
+end subroutine
+
+subroutine test_parallel_loop_with_reduction
+  integer :: i, total
+  total = 0
+  !PARSE-TREE: OmpBeginLoopDirective
+  !PARSE-TREE-NEXT: OmpDirectiveName -> llvm::omp::Directive = parallel loop
+  !CHECK: !$omp parallel loop reduction(+: total)
+  !$omp parallel loop reduction(+:total)
+  do i=1,10
+   total = total + i
+  end do
+end subroutine
+
+subroutine test_parallel_loop_with_if
+  integer :: i, j = 1
+  logical :: cond
+  cond = .true.
+  !PARSE-TREE: OmpBeginLoopDirective
+  !PARSE-TREE-NEXT: OmpDirectiveName -> llvm::omp::Directive = parallel loop
+  !CHECK: !$omp parallel loop if(cond) proc_bind(close)
+  !$omp parallel loop if(cond) proc_bind(close)
+  do i=1,10
+   j = j + 1
+  end do
+end subroutine

